lc-hax/Scripts/Patches/GodModePatch.cs

```diff
@@ -1,10 +1,123 @@
-#pragma warning disable IDE1006
-
 using GameNetcodeStuff;
 using HarmonyLib;
 using Hax;
+using UnityEngine;
 
-[HarmonyPatch(typeof(PlayerControllerB), nameof(PlayerControllerB.AllowPlayerDeath))]
+[HarmonyPatch]
 class GodModePatch {
-    static bool Prefix(PlayerControllerB __instance) => !Setting.EnableGodMode || Helper.LocalPlayer?.actualClientId != __instance.actualClientId;
+    [HarmonyPrefix]
+    [HarmonyPatch(typeof(PlayerControllerB), nameof(PlayerControllerB.AllowPlayerDeath))]
+    static bool PrefixDamagePlayer(PlayerControllerB __instance, ref bool __result) {
+        if (__instance.isSelf())
```

Comment:
if-guard

Reply:
is THE SAME!

Reply:
just because the logic is the same, doesn’t mean it’s good. nesting forces the user to read the conditional when otherwise, it doesn’t have to be read.